src/rules/NoUnknownInputTypes.js

```diff
@@ -0,0 +1,35 @@
+var _error = require("graphql/error");
+
+function noUnknownInputTypeMessage(operationName, inputType) {
+  return `${operationName} has unknown input type: ${inputType}`;
+}
+
+/**
+ * No unknown input types
+ *
+ * A query and/or mutation should have a defined input type if not a primitive
+ */
+
+function NoUnknownInputTypes(context) {
+  return {
+    OperationDefinition(node) {
+      if (node.variableDefinitions.length > 0) {
+        node.variableDefinitions.forEach(def => {
+          if (def.type.type) {
+            const type = def.type.type;
+            if (!context._schema._typeMap[type.name.value]) {
+              context.reportError(
+                new _error.GraphQLError(
+                  noUnknownInputTypeMessage(node.name.value, type),
```

Comment:
should this be using type.name.value?

Reply:
Maybe not. What do you suggest?

Reply:
@Viper512 (Corey G.) pulled in the latest `graphql-mask` and ran into another issue. I'm not sure if its related to this change, or something else (we still have an issue with the mutations, but a different error this time). 

I think he's been fiddling around with the code in `graphql-mask` to see if he could get it to work, but I don't know that we've verified anything against the current unit tests.  We probably need to dig a little deeper.

Reply:
with node.name.value, i'm receiving the following error : 
(node:11740) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): TypeError: Cannot read property 'value' of undefined
(node:11740) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

With type.name.value, it's working fine for me.

Which makes me think that Line 23 has a typo on it.

Reply:
Hmm.. I feel like every operation should have a name, but could be wrong. Same goes for types.

What do the query and schema look like? Can you provide a reproduction?